src/object/schema_manager.c

```diff
@@ -15883,26 +15881,18 @@ sm_truncate_class_internal (MOP class_mop)
 	}
     }
 
-#if 0
-  if (keep_pk == true && log_does_allow_replication () == true)
-    {
-      error = sm_truncate_using_delete (class_mop);
-    }
-  else
+  if (sm_is_reuse_oid_class (class_mop))
     {
       error = sm_truncate_using_destroy_heap (class_mop);
     }
-  if (error != NO_ERROR)
+  else
     {
-      goto error_exit;
+      error = sm_truncate_using_delete (class_mop);
```

Comment:
In this case, The locator_remove_class_from_index() can be called. Why is it different from before?

Reply:
No matter an index is droppable or not, the heap records in a heap with DONT_REUSE_OID have to be deleted record by record. I'll be modified when dealing with the type of heap file.